### ISSUE TITLE
conan config install https://... fails when conan_server's certificate is self-signed

### ISSUE THREAD

**Issue by matlo607** [conan-io/conan#2460]
Hello,

I am using conan 1.0.4 and experienced an issue while I was trying to install a configuration for a conan client ([\[docs.conan.io\] conan config install](http://docs.conan.io/en/latest/reference/commands/consumer/config.html)).
This server has a self-signed certificate.
```
$ conan config install https://conan_server/conan-client/config/conan-default-config.zip
Trying to download  https://conan_server/conan-client/config/conan-default-config.zip
ERROR: Error downloading file https://conan_server/conan-client/config/conan-default-config.zip: 'HTTPSConnectionPool(host='conan_server', port=443): Max retries exceeded with url: /conan-client/config/conan-default-config.zip (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')],)",),))'
Waiting 5 seconds to retry...
ERROR: Error downloading file https://conan_server/conan-client/config/conan-default-config.zip: 'HTTPSConnectionPool(host='conan_server', port=443): Max retries exceeded with url: /conan-client/config/conan-default-config.zip (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')],)",),))'
```
`conan config install` does not have currently an option to disable the verification of the certificate.
The download is performed in [conans/client/conf/config_installer.py#L83](https://github.com/conan-io/conan/blob/8c0b516f3a13efe68144d146d86b9ba80c213624/conans/client/conf/config_installer.py#L83) using `tools.download()`. This utility has a parameter to enable/disable ssl checks.
```python
def download(url, filename, verify=True, out=None, retry=2, retry_wait=5, overwrite=False,
             auth=None, headers=None)
```
I think a flag to disable ssl check would be a good solution in this use case.
What do you think about it ?
Is there currently another solution to perform the download from a server with a self-signed certificate ?

EDIT:
Someone running a git repository with a sefl-signed certificate will experience the same issue.
```python
def _process_git_repo(repo_url, client_cache, output, runner, tmp_folder):
    output.info("Trying to clone repo  %s" % repo_url)

    with tools.chdir(tmp_folder):
        runner('git clone "%s" config' % repo_url, output=output)
    tmp_folder = os.path.join(tmp_folder, "config")
    _process_folder(tmp_folder, client_cache, output)
```
I guess this case can be solved by setting manually `GIT_SSL_NO_VERIFY` to `true`.

**Comment by memsharded** [comment]
Hi!

Thanks for reporting this issue, indeed it seems a problem.

Could you please try hardcoding ``verify=False`` in https://github.com/conan-io/conan/blob/8c0b516f3a13efe68144d146d86b9ba80c213624/conans/client/conf/config_installer.py#L83, to check that it really solves your issue? Then we could think of exposing that as an argument.

Another approach would be trying to add your own self-signed certificate to the ``<userhome>/.conan/cacert.pem`` file. I understand this is manual, and then not very convenient, but would keep the SSL checks active.

**Comment by matlo607** [comment]
Hardcoding `verify=False` worked.
```
$ conan config install https://conan_server/config
Trying to download  https://conan_server/config
/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py:858: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
[==================================================] 2.9KB/2.9KB      
Unzipping 5.0KB
Unzipping 100 %                                                       Installing settings.yml
Defining remotes
WARN: Remotes registry file missing, creating default one in /home/matt/.conan/registry.txt
Installing profiles
    Installing profile rhel-gcc49
Processing conan.conf
```

**Comment by memsharded** [comment]
Ok, thanks for telling, looks good. We will have a look and discuss the UI for this opt-in, I guess something like ``--verify=False`` in command line, though it looks a bit ugly. Lets think about it.

Would be the other approach viable for you? Adding your certificate to the ``cacert.pem`` file? Thanks!

**Comment by matlo607** [comment]
I am not in favor of `cacert.pem` because the user has to perform one more operation to download and install this file whereas the purpose of `conan config install` is to setup everything in user's conan home automatically in one shot.

`wget` has `--no-check-certificate` that is explicit, `curl` has `--insecure` that is a bit obscure from my point of view, I would propose `--ssl-verify=False` so that the relation with ssl is clear.

Thanks for your help and reactivity.

**Comment by lasote** [comment]
For me the `--ssl-verify=False` as a parameter of the `conan config install` is ok. I agree that it is a very common argument in some client tools.

**Comment by memsharded** [comment]
Ok, then. Lets do ``ssl-verify``.
I have assigned it to @danimtb for 1.2, but it shouldn't be difficult to implement, so if possible to speed it up to 1.1, we might try.

**Comment by memsharded** [comment]
It has been merged, so it will be in next conan 1.1. Thanks!
